firebase-appdistribution/test-app/src/main/java/com/googletest/firebase/appdistribution/testapp/NotificationFeedbackTrigger.kt

```diff
@@ -0,0 +1,215 @@
+import java.io.IOException
+
+@SuppressLint("StaticFieldLeak") // Reference to Activity is set to null in onActivityDestroyed
+object NotificationFeedbackTrigger : Application.ActivityLifecycleCallbacks {
+    private const val TAG: String = "NotificationFeedbackTrigger"
+    private const val FEEBACK_NOTIFICATION_CHANNEL_ID = "InAppFeedbackNotification"
+    private const val FEEDBACK_NOTIFICATION_ID = 1
+    const val SCREENSHOT_FILE_NAME =
+        "com.googletest.firebase.appdistribution.testapp.screenshot.png"
+
+    private var isEnabled = false
+    private var hasRequestedPermission = false
+    private var currentActivity: Activity? = null
+    private var requestPermissionLauncher: ActivityResultLauncher<String>? = null
+
+    fun initialize(application: Application) {
+        // Create the NotificationChannel, but only on API 26+ because
+        // the NotificationChannel class is new and not in the support library
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            val channel = NotificationChannel(
+                FEEBACK_NOTIFICATION_CHANNEL_ID,
+                application.getString(R.string.feedback_notification_channel_name),
+                NotificationManager.IMPORTANCE_HIGH
+            )
+            channel.description =
+                application.getString(R.string.feedback_notification_channel_description)
+            val notificationManager = application.getSystemService(
+                NotificationManager::class.java
+            )
+            notificationManager.createNotificationChannel(channel)
+        }
+        application.registerActivityLifecycleCallbacks(this)
+    }
+
+    fun enable(currentActivity: Activity? = null) {
+        this.currentActivity = currentActivity
+        isEnabled = true
+        if (currentActivity != null) {
+            showNotification(currentActivity)
+        }
+    }
+
+    fun disable() {
+        isEnabled = false
+        val activity = currentActivity
+        currentActivity = null
+        if (activity != null) {
+            cancelNotification(activity)
+        }
+    }
+
+    private fun showNotification(activity: Activity) {
+        if (ContextCompat.checkSelfPermission(activity, POST_NOTIFICATIONS) == PERMISSION_GRANTED) {
+            val intent = Intent(activity, TakeScreenshotAndTriggerFeedbackActivity::class.java)
+            intent.addFlags(Intent.FLAG_ACTIVITY_NO_HISTORY)
+            val pendingIntent =
+                PendingIntent.getActivity(activity, 0, intent, PendingIntent.FLAG_IMMUTABLE)
+            val builder = NotificationCompat.Builder(activity, FEEBACK_NOTIFICATION_CHANNEL_ID)
+                .setSmallIcon(R.drawable.test_app_icon)
+                .setContentTitle(activity.getText(R.string.feedback_notification_title))
+                .setContentText(activity.getText(R.string.feedback_notification_text))
+                .setPriority(NotificationCompat.PRIORITY_HIGH)
```

Comment:
Should this be low priority?

Reply:
See above. I just want to make sure it's shown. I'll be up to the developer(s) adopting this code to select an appropriate priority.